openpype/hosts/blender/scripts/build_workfile.py

```diff
@@ -874,6 +875,120 @@ def build_lipsync(project_name: str, shot_name: str):
             )
 
 
+def build_fabrication(project_name: str, asset_name: str):
+    """Build fabrication workfile.
+
+    Args:
+        project_name (str): The current project name from OpenPype Session.
+        asset_name (str): The current asset name from OpenPype Session.
+    """
+
+    # Get subsets
+    asset_doc = get_asset_by_name(
+        project_name, "LightSetupBank", fields=["_id"]
+    )
+    subsets = get_subsets(
+        project_name,
+        asset_ids={asset_doc["_id"]},
+        fields=["_id", "name", "data"],
+    )
+
+    # Match and download and load subset
+    for subset in subsets:
+        raw_asset_name = asset_name.split("_")[0].lower()
+        raw_subset_name = (
+            subset["name"].strip("lighting").rsplit("_", 1)[0].lower()
+        )
+        if raw_asset_name == raw_subset_name:
+            representation = download_subset(
+                project_name,
+                "LightSetupBank",
+                subset["name"],
+            )
+
+    # Wait for downloads to be finished
+    wait_for_download(project_name, [representation])
+    # Load representation
+    load_subset(project_name, representation, "AppendBlenderLightingLoader")
+
+    # Create setdress instance
+    bpy.ops.scene.create_openpype_instance(
+        creator_name="CreateWoollySetdress",
+        asset_name=asset_name,
+        subset_name="setdressMain",
+        gather_into_collection=True,
+    )
+
+    # Store setdress instance
+    setdress_collection = list(
+        bpy.context.scene.openpype_instances[-1].get_root_outliner_datablocks()
+    )[-1]
+
+    # Store camera name for camera instance
+    camera_name = f"{asset_name}_cameraMain"
+
+    # Create camera instance
+    # If `camera_name` exists don't create it
+    bpy.ops.scene.create_openpype_instance(
+        creator_name="CreateCamera",
+        asset_name=asset_name,
+        subset_name="cameraMain",
+        gather_into_collection=True,
+        datapath="objects",
+        datablock_name=camera_name,
```

Comment:
If we are designing a whole build, with the camera creation, we shouldn't have to pass a camera name here, because there is none in the scene. Both `datapath` and `datablock_name` are unrelevant here.


Therefore, you'll have to get the created camera after and use it.